### PR TITLE
Removal expr node

### PR DESCRIPTION
**Pull request by yliuuuu**
## Relevant Issues
- [Related To] #682 
- [Related To] #878 

## Description
- Removes the deprecated ExprNode and AstNode.

## Other Information
- Updated Unreleased Section in CHANGELOG: **[YES/NO]**
  - Yes. 
- Any backward-incompatible changes? **[YES/NO]**
  - Yes, the deprecated ExprNode and AstNode are removed and no longer supported. 
  - < For this purpose, we define backward-incompatible changes as changes that—when consumed—can potentially result in
errors for users that are using our public APIs or the entities that have `public` visibility in our code-base. >
- Any new external dependencies? **[YES/NO]**
  - No. 

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/design/Abstract Syntax Tree.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# This Doc covers the ExprNode AST, which has been deprecated and removed from the Codebase. 
+
```
Comment (lziq) [inline docs/design/Abstract Syntax Tree.md:2]: I think we need to see if I want to remove this file or change this file so it introduces PIG AST.

File: lang/src/org/partiql/lang/ast/SqlDataType.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ *  language governing permissions and limitations under the License.
+ */
+
+package org.partiql.lang.ast
+
+/**
+ * The core PartiQL data types.
+ * TODO: Remove this once SqlParser has been removed from this codebase.
+ */
+sealed class SqlDataType(val typeName: String, open val arityRange: IntRange) {
```
Comment (lziq) [inline lang/src/org/partiql/lang/ast/SqlDataType.kt:21]: I think this represents data types for `ExprNode`. Are we able to remove it?

File: lang/src/org/partiql/lang/ast/meta.kt [hunk 3]
```diff
@@ -69,3 +67,5 @@ fun MetaContainer.find(tagName: String): Meta? = this[tagName] as Meta?
 fun MetaContainer.hasMeta(tagName: String) = this.containsKey(tagName)
 
 fun MetaContainer.add(meta: Meta): MetaContainer = this.plus(metaContainerOf(meta))
+
+internal typealias IonElementMetaContainer = com.amazon.ionelement.api.MetaContainer
```
Comment (lziq) [inline lang/src/org/partiql/lang/ast/meta.kt:71]: Why can't we directly use `com.amazon.ionelement.api.MetaContainer`?

File: lang/test/org/partiql/lang/ast/PathComponentExprTest.kt [hunk 4]
```diff
@@ -14,111 +14,100 @@
...
-            TestCase(oneSensitive, oneSensitive.copy(expr = litInt(2)), false),
-            TestCase(oneInsensitive, oneInsensitive.copy(expr = litInt(2)), false),
-
-            // isn't equivalent because PathComponentExpr.case doesn't match (but [PathComponentExpr.expr] does.)
-            TestCase(fooSensitive, fooInsensitive, false),
-            TestCase(fooSensitive, oneInsensitive, false),
-            TestCase(oneSensitive, oneInsensitive, false)
-        )
-}
+// TODO: This is a test for node equivalence for the deprecated ExprNode.It has not been removed because of this issue: https://github.com/partiql/partiql-lang-kotlin/issues/878
```
Comment (lziq) [inline lang/test/org/partiql/lang/ast/PathComponentExprTest.kt:17]: Did we actually get into a conclusion that equivalence of AST nodes is not useful?

Comment (lziq) [reply lang/test/org/partiql/lang/ast/PathComponentExprTest.kt:17]: If so, we can remove this file

File: lang/test/org/partiql/lang/ast/VariableReferenceTest.kt [hunk 5]
```diff
@@ -14,63 +14,58 @@
...
-    }
-
-    @Test
-    fun caseInsensitiveNotEquals() {
-        assertNotEquals(insensitiveFoo, sensitiveFoo)
-        assertNotEquals(insensitiveFoo, insensitiveFoo.copy(scopeQualifier = ScopeQualifier.LEXICAL))
-        assertNotEquals(insensitiveFoo, insensitiveFoo.copy(metas = metaContainerOf(SourceLocationMeta(1, 1))))
-    }
-}
+// TODO: This is a test for node equivalence for the deprecated ExprNode.It has not been removed because of this issue: https://github.com/partiql/partiql-lang-kotlin/issues/878
```
Comment (lziq) [inline lang/test/org/partiql/lang/ast/VariableReferenceTest.kt:17]: Same here

File: CHANGELOG.md [hunk 6]
```diff
@@ -56,7 +56,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   - [AstVisitor] Interface & [AstVisitorBase] class
   - [AstWalker] class
   - [MetaStrippingRewriter] class
-- Removes the deprecated SqlParser and SqlLexer 
+- Removes the deprecated ExprNode and related files in the code base. 
+  - [Parser] API `parseExprNode(source: String): ExprNode` has been removed.
+  - [CompilerPipeline] API `compile(query: ExprNode): Expression` has been removed.
+  - [ExprNode] and [AstNode] have been removed.
+  - Functions related to conversions between ExprNode and PartiqlAst have been removed. 
+  - Parser Tests no longer do round trip.
```
Comment (lziq) [inline CHANGELOG.md:64]: Do we need to mention tests in the CHANGELOG?

File: lang/test/org/partiql/lang/ast/PathComponentExprTest.kt (hunk 4 above)
Comment (yliuuuu) [reply lang/test/org/partiql/lang/ast/PathComponentExprTest.kt:17]: It is not used as of now in the library, but I will not say it is not useful. See https://github.com/partiql/partiql-lang-kotlin/issues/878

File: docs/design/Abstract Syntax Tree.md (hunk 1 above)
Comment (yliuuuu) [reply docs/design/Abstract Syntax Tree.md:2]: Probably worth another PR to clean up the documentation in the code base. Issue is created: https://github.com/partiql/partiql-lang-kotlin/issues/879

**Review by lziq — APPROVED** [review]
Looks good to me

**Pull request merged**